modules/core/src/main/java/org/apache/synapse/config/xml/ProxyServiceSerializer.java

```diff
@@ -239,7 +239,10 @@ public static OMElement serializeProxy(OMElement parent, ProxyService service) {
             parent.addChild(proxy);
         }
 
-        CommentListUtil.serializeComments(proxy, service.getCommentsList());
+        List<String> commentsList = service.getCommentsList();
```

Comment:
In ProxyService the  commentsList is initialized to an empty ArrayList. Therefore can  `service.getCommentsList()` return a null?